### PR TITLE
FOGL-9177 Missing import statement fixes on plugin start & also error handling …

### PR DESCRIPTION
**Pull request by ashish-jabble**
…when message payload is differ than existing sparkplug proto

### CODE REVIEW

**Review by MarkRiddoch — APPROVED** [review]

**Review by MarkRiddoch — APPROVED** [review]

File: python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py [hunk 1]
```diff
@@ -215,41 +220,61 @@ def plugin_register_ingest(handle, callback, ingest_ref):
...
+        time_stamp = utils.local_timestamp()
+        try:
+            string_data = msg.payload.decode('utf-8')
+            dict_data = json.loads(string_data)
+            data = {
+                'asset': _assetName,
+                'timestamp': time_stamp,
+                'readings': dict_data if isinstance(dict_data, dict) else {"value": string_data}
+            }
+            _callback_event_loop.run_until_complete(save_data(data))
```
Comment (praveen-garg) [inline python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py:257]: We should not introduce this.

Payload should be spBv1.0 complaint only.

File: python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py [hunk 2]
```diff
@@ -215,41 +220,61 @@ def plugin_register_ingest(handle, callback, ingest_ref):
...
+        except UnicodeDecodeError:
+            # Serialized data case with protobuf
+            inbound_payload = sparkplug_b_pb2.Payload()
+            inbound_payload.ParseFromString(msg.payload)
+            for metric in inbound_payload.metrics:
+                data = {
+                    'asset': metric.name,
+                    'timestamp': time_stamp,  # metric.timestamp
+                    'readings': {
+                        "value": metric.float_value,
```
Comment (praveen-garg) [inline python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py:267]: For sinusoid via MQTT north sparkplug B this should be enough to bring it back to life.

However, We should target a minimum of in next branch

```
# Define the message payload with different metric types
message = {
    "metrics": [
        {
            "name": "temperature",
            "value": 22.5,  # Float
            "timestamp": int(time.time()),
            "type": "float"
        },
        {
            "name": "humidity",
            "value": 55,  # Integer
            "timestamp": int(time.time()),
            "type": "integer"
        },
        {
            "name": "status",
            "value": "active",  # String
            "timestamp": int(time.time()),
            "type": "string"
        },
        {
            "name": "powerOn",
            "value": True,  # Boolean
            "timestamp": int(time.time()),
            "type": "boolean"
        }
    ]
}
```

May be we can also support float array, dict which Fledge supports.

File: python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py [hunk 3]
```diff
@@ -215,41 +220,61 @@ def plugin_register_ingest(handle, callback, ingest_ref):
...
+        except Exception:
+            # pass message payload as is
             data = {
-                'asset': metric.name,
-                'timestamp': time_stamp,  # metric.timestamp
+                'asset': _assetName,
+                'timestamp': time_stamp,
                 'readings': {
-                    "value": metric.float_value,
+                    "value": string_data,
```
Comment (praveen-garg) [inline python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py:277]: We should use metric name in reading name value pair?

File: python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py (hunk 2 above)
Comment (ashish-jabble) [reply python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py:267]: We have another JIRA FOGL-9198 to map Fledge Support types

**Comment by ashish-jabble** [comment]
This PR only meant for to add south bound service without any errors. Others issues will be handled by others PR/JIRA

**Pull request merged**